### ISSUE TITLE
Crash free users metric not populating with manual instrumentation

### ISSUE THREAD

**Issue by TheDuckCow** [getsentry/sentry#56895]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

The root of my issue is that while we are successfully viewing crash free session rates in the project overview, the crash free users stat is not populating, despite us including users in requests.

My team and I are building plugins for 3d software written in python. We have to use manual instrumentation, otherwise we start capturing data and exceptions from the host application or other plugins unrelated to us (both unuseful, and a privacy issue).

We manually start and end transactions around any user interaction. If we receive an unhandled exception, we pipe that to sentry and call this function we have written to set the overall session as crashed (this is working in one of our projects, but not the other):

```
def _set_session_crashed() -> None:
    """Forces the session to appear as crashed.

    Had to pull this from lower level sources of the sentry lib to assign, see
    Hub.py's start_session function for partial reference, and session.py
    for updating aspects of the session variable itself.
    """
    hub = sentry_sdk.Hub.current
    client, scope = hub._stack[-1]
    if scope._session:
        scope._session.update(status="crashed")
```

I realize we're pulling from some private internals so I'm happy to be told there's a different way here to manually set a session as crashed, but all the same we haven't figured out why in one project the stats are populating and in the other it isn't.

### Expected Result

This is from one of our plugin projects where crash free users _is_ populating. Notice how both crash free sessions and crash free users at the top are populated.

<img width="840" alt="Screen Shot 2023-09-25 at 3 08 07 PM" src="https://github.com/getsentry/sentry/assets/2958461/d14a9eaa-9fa8-4462-a2d5-7213c7d32602">

### Actual Result

This is from another project, which from the best that I can tell is set up the same way, but crash-free users are not populating (even though session health and crash free sessions are populated).

<img width="842" alt="Screen Shot 2023-09-25 at 3 07 22 PM" src="https://github.com/getsentry/sentry/assets/2958461/f7df40d9-eb8b-48c9-928b-5f2f4394a955">

See here proof that we are indeed also capturing users:

<img width="694" alt="Screen Shot 2023-09-25 at 3 15 50 PM" src="https://github.com/getsentry/sentry/assets/2958461/0bd1f1c5-6daa-4931-9a91-482dda54b772">

...and how it's recognized in other interface areas:

<img width="308" alt="Screen Shot 2023-09-25 at 3 15 16 PM" src="https://github.com/getsentry/sentry/assets/2958461/4ddf03bd-1696-4c12-a19d-4931238befcb">

### Product Area

Dashboards

### DSN

https://c1e724fdb8ad4cc1b96b7cd6b79d8a3e@o1010513.ingest.sentry.io/4504062502764544

**Comment by narsaynorath** [comment]
hey @TheDuckCow, thanks for the report. I just wanted to let you know that we're investigating this. We'll add updates here when we find out more/need more information

**Comment by TheDuckCow** [comment]
Thank you @narsaynorath! Keen to hear whatever advice you have to share. It's very possible we're doing something not right on our side, again given our manual instrumentation, so happy to try any troubleshooting steps if you can think of any.

**Comment by jan-auer** [comment]
In short, Release Health for Python does not support crash-free users.

Release Health has two modes for reporting sessions:
 - **Application Mode Sessions:** This mode is used for user-attended programs, which typically have a single long running session that span the application's lifetime.
- **Request Mode Sessions:** This mode is used in servers that report one session per incoming request, and thus have a lot of very short lived sessions.

Request mode sessions report their data in batches without user or device IDs, which is why crash free users will not show data on these platforms. The default mode for Python web servers is this request mode.

There are currently no active plans to add crash free users for server platforms, although we are working on a technical foundation that would make this possible in the future. We'd certainly love to hear more if this feature is valuable to you how you would incorporate this information in your workflows.

**Comment by TheDuckCow** [comment]
Hi @jan-auer thanks for the information. That's actually all I needed to know, that it's not working in request mode - we've got it working now!

We had inadvertently set the mode to request, whereas our other python repo is set to application (not sure how we flipped that one around). Now that I've made that change, we are getting the crash free user reporting that we were hoping for:

<img width="866" alt="Screen Shot 2023-10-04 at 12 47 42 PM" src="https://github.com/getsentry/sentry/assets/2958461/9c7c1a51-2b69-4925-b46f-4865f43f31e8">

That being said, I hear what you are saying and am acutely aware that we are essentially relying on unsupported behavior. So you are aware, the way we have manual instrumentation working in the above python project:
- Set the mode to application mode, even though we are essentially treating it as request mode.
- We are manually starting/ending sessions around individual user operations. So we have it set up so we do have a large number of short events, instead of a single long running session (since our plugin is really a guest inside of another pre-existing host application, and we don't want to overreport on data). 
- We do include unique user id's with each of these sessions/requests, this unique id does not change between requests on the same user's system.
- We have a function `_set_session_crashed` mentioned in my first post which is used to identify a session (ie single operation) as a failure that we want our "crash free rate" to consider as a drop. We call this function whenever we want to treat a situation as a "crash".

In terms of our use case and why this metric is helpful: basically we needed a way to be able to measure what % of users are not running into critical problems. With manual instrumentation, we happen to have nice control over defining this metric (basically in our specific case, any unhandled exception that falls back to our catch-all sentry wrapper -> define as a "crash"). If crash-free sessions are a high %, it doesn't necessarily tell the story if there's one kind of request that is failing _for every user_ which would more only show up in the crash free user report. It's a better holistic view when talking about health for request mode sessions when considering what typical users are experiencing. But because of the host application structure and shared python environment, we couldn't use the typical setup of application mode reporting. Again, the use is not actually a server, but a local setup running on a user's machine within another pre-existing desktop application. So I get that our use case here is a bit strange.

Anyhow, this has resolved the issue from our standpoint, thanks again!
